scripts/create_placeholder_users.py

```diff
@@ -0,0 +1,92 @@
+python manage.py runscript create_placeholder_users --script-args http://localhost:8080/api/v1 username password
+Run me in production:
+python manage.py runscript create_placeholder_users --script-args http://girder.dandiarchive.org/api/v1 username password
+
+Only dandisets that exist in both Girder and Django are considered, all others are ignored.
+This is determined by matching the Girder dandiset identifier with the identifier stored in the Django metadata.
+Once dandisets are matched, the script will do the following for each Girder owner:
+ * Check if a Django user exists with the Girder owner's email
+ * If not, check if a Django user exists with the Girder owner's email, prepended with 'placeholder_'
+ * If not, create a placeholder user in Django
+It will then assign ownership of the dandiset to the user.
+This prevents placeholders from being created for users who already exist or have already been placeholdered.
+"""
+from girder_client import GirderClient
+from guardian.shortcuts import assign_perm
+from dandiapi.api.models import Dandiset, Version
+from django.contrib.auth.models import User
+
+
+def get_girder_dandisets(gc):
+    dandisets = {}
+    for dandiset in gc.get('/dandi'):
+        identifier = dandiset['meta']['dandiset']['identifier']
+        dandisets[identifier] = [
+            user['id'] for user in dandiset['access']['users'] if user['level'] == 2
+        ]
+    return dandisets
+
+
+def find_or_find_placeholder_or_create_placeholder(girder_user):
+    email = girder_user['email']
+    try:
+        return User.objects.get(email=email)
+    except User.DoesNotExist:
+        pass
+    try:
+        return User.objects.get(email=f'placeholder_{email}')
+    except User.DoesNotExist:
+        print(f'Creating new placeholder user placeholder_{email}')
+        user = User(
+            username=girder_user['login'],
+            first_name=girder_user['firstName'],
+            last_name=girder_user['lastName'],
+            email=f'placeholder_{email}',
+        )
+        user.save()
+        return user
+
+
+def migrate_dandiset(gc, dandiset, user_ids):
+    for user_id in user_ids:
+        girder_user = gc.get(f'user/{user_id}')
+        django_user = find_or_find_placeholder_or_create_placeholder(girder_user)
+        if not django_user.has_perm('owner', dandiset):
+            print(f'Assigning ownership of {dandiset.identifier} to {django_user}')
+            assign_perm('owner', django_user, dandiset)
+
+
+def migrate(gc):
+    girder_dandisets = get_girder_dandisets(gc)
+    for dandiset_identifier in girder_dandisets:
+        # Do not use the dandiset identifier, it is an autoincremented primary key
+        # and cannot be trusted.
+        # We are also assuming that no two dandisets will have metadata with the same identifier.
+        # TODO make sure this query works with all identifier formats
+        version = Version.objects.filter(metadata__metadata__identifier=dandiset_identifier).first()
```

Comment:
I assume that, by design, `version` would be either an empty list or a singleton list. Is there any condition that could lead to multiple versions being found here?

Reply:
This finds an arbitrary version which has metadata that contains a dandiset identifier. There could hypothetically be multiple published versions to match against.

It's ultimately OK if there are multiple versions, since we are actually only using `version.dandiset`, which should hopefully be the same for all versions which claim to belong to a dandiset.